golem-component-service-base/src/repo/component.rs

```diff
@@ -478,9 +701,209 @@ impl ComponentRepo for DbComponentRepo<sqlx::Postgres> {
         Ok(())
     }
 
+    async fn update(
+        &self,
+        owner: &Owner::Row,
+        namespace: &str,
+        component_id: &Uuid,
+        data: Vec<u8>,
+        metadata: Vec<u8>,
+        component_type: Option<i32>,
+        files: Option<Vec<FileRecord>>,
+    ) -> Result<ComponentRecord, RepoError> {
+        let mut transaction = self.db_pool.begin().await?;
+
+        let result = sqlx::query("SELECT namespace FROM components WHERE component_id = $1")
```

Comment:
i would probably make this part of the update through a join, and make them a predicate.
As for the "WITH prev selects", i think those are not reliable at all without golding explicit locks for the affected rows (or use a pattern where before any component version changes we do a single row lock on the components table)